nixos/modules/services/system/cloud-init.nix

```diff
@@ -3,7 +3,16 @@
 with lib;
 
 let cfg = config.services.cloud-init;
-    path = with pkgs; [ cloud-init nettools utillinux e2fsprogs shadow openssh iproute ];
+    path = with pkgs; [
+      btrfs-progs
```

Comment:
`btrfs-progs` adds like `3.5mb`. I think we can add an option called `btrfs.enable` (default: false) and an option `ext4.enable` (default: true).

Reply:
@Mic92 Sure. Sounds like a good idea to me.

I will re-write this as a module in the coming commits of this PR.